src/components/Trade/Trade.tsx

```diff
@@ -30,15 +32,35 @@ export const Trade = ({ defaultBuyAssetId }: TradeProps) => {
     },
   })
 
+  // The route has changed, so re-enable the default values useEffect
+  useEffect(() => setHasSetDefaultValues(false), [location])
+
   useEffect(() => {
+    if (hasSetDefaultValues) return
     ;(async () => {
       const result = await getDefaultAssets()
       if (!result) return
       const { buyAsset, sellAsset } = result
       methods.setValue('sellTradeAsset.asset', sellAsset)
       methods.setValue('buyTradeAsset.asset', buyAsset)
+      const defaultAssetsAreChainDefaults =
```

Comment:
This is a bit hacky, and I dislike it. 

If the default assets we get back from the result aren't just the defaults, then the hook had enough information to set the assets to something else.

But Apotheosis, what if we were already on the default assets and this fired? 
You're right, we'd re-set them - but the user would observe no change (I think).